### PR TITLE
New COBOL85 source format and preliminary edits before introducing Area A enforcement

### PR DESCRIPTION
**Pull request by nberth**

### CODE REVIEW

File: cobc/cobc.h [hunk 1]
```diff
@@ -95,14 +95,17 @@
 /* Source format enum */
 enum cb_format {
 	CB_FORMAT_FIXED = 0,
+	CB_FORMAT_COBOL85,	/* Fixed format with Area A checks (pending) */
```
Comment (GitMensch) [inline cobc/cobc.h:98]: While we're here... both CB_FORMAT_FIXED and CB_FORMAT_VARIABLE should get a comment referencing COBOL 2002+ fixed-form/free-form and the comment for the enum changed to "Source reference-format enum"

**Review by GitMensch — CHANGES_REQUESTED** [review]
Looks good to me in general.

apart from the comments:
* the new fformat should be used in cobol85, xopen and likely mvs,bs2000,gcos and possibly rm,realia,ibm
* help.c should have been adjusted before for --fixed / --free to note their aliasing to `-fformat`

**Comment by nberth** [comment]
> * the new fformat should be used in cobol85, xopen and likely mvs,bs2000,gcos and possibly rm,realia,ibm

The issue is that the `format` option is not a dialect option. May I define it as such?

File: po/fr.po [hunk 2]
```diff
@@ -2062,12 +2062,18 @@ msgstr ""
 "                        voyez les fichiers de configuration dans le répertoire de configuration"
 
 #: cobc/help.c:82
-msgid "  -F, -free             use free source format"
-msgstr "  -F, -free             utiliser le format source libre"
+msgid ""
```
Comment (GitMensch) [inline po/fr.po:2065]: The idea is good, but we don't do the changes directly, they would be overwritten.

Please target any updates to https://translationproject.org/domain/gnucobol.html. As those are new / changed msgids the translation will need to wait until the release candidate (End of August) is done and the new gnucobol.pot is sent to the TP (at this time we'll pull all current translations, then let the files be recreated, send the msgid file to TP - and before the release pull the translations again).

**Review by GitMensch — APPROVED** [review]
that's fine - please commit upstream *without fr.po*

**Comment by GitMensch** [comment]
> > * the new fformat should be used in cobol85, xopen and likely mvs,bs2000,gcos and possibly rm,realia,ibm
> 
> The issue is that the `format` flag is not a dialect option. May I define it as such? We've discussed that at some point, but maybe we needed the flag out to finally decide that.

Oops missed that - yes, it should be moved from flag.def to config.def - please do as part of this PR!
Note: standard cobol still has "The default reference format of a compilation group is fixed-form." - so no big breaking change for default / -fixed `-std=cobolNNN`, a minor for `-std=cobol85` and the other "old" ones - but that's more a correction.

**Review by GitMensch — CHANGES_REQUESTED** [review]
... please move the default format out to a dialect option (the way to hint at that is already in the NEWS, you'd move the -fformat entry and say that's the dialect option).

File: po/fr.po (hunk 2 above)
Comment (nberth) [reply po/fr.po:2065]: Ok noted. I was not really aware of that process for translations. Thanks for the info :smile:

File: NEWS [hunk 3]
```diff
@@ -38,6 +38,18 @@ NEWS - user visible changes				-*- outline -*-
    this isn't done anymore so if you need the memory to be initialized
    specify that explicit in the source and recompile
 
+** new -fformat dialect option, and extended SOURCE FORMAT directives, with
+   the following newly supported reference-formats (in addition to
+   FIXED/FREE):
+
+   COBOL85              Fixed-form format with enforcement of Area A (pending)
+   VARIABLE		Microfocus Variable Fixed-form format
```
Comment (GitMensch) [inline NEWS:46]: that would be "Micro Focus"; please change that list to be in alphabetical order

File: cobc/ChangeLog [hunk 4]
```diff
@@ -1,4 +1,14 @@
 
+2022-07-28  Nicolas Berthier <nicolas.berthier@ocamlpro.com>
+
+	* flag.def, config.def: make format a dialect option instead of a flag
+	* help.c: mention alias for options --free and --fixed
+	* cobc.h: extend cb_format_enum with COBOL85 (fixed format with
+	  pending Area A enforcement); define CB_SF_FIXED and CB_SF_FREE
+	* cobc.c: use new source format macros, whitespace fixes
+	* parser.y, pplex.l: annot every token that is not used in the grammar,
```
Comment (GitMensch) [inline cobc/ChangeLog:9]: /annot/annotate/ - no use in mentioning the whitespace fixes (not relevant for copyright or future programmers)

File: cobc/config.def [hunk 5]
```diff
@@ -79,6 +79,9 @@ CB_CONFIG_ANY (signed int, cb_default_byte, "defaultbyte",
 
 CB_CONFIG_ANY (enum cb_std_def, cb_std_define, "standard-define", NULL) /* Used Standard (set via -std) */
 
+CB_CONFIG_ANY (enum cb_format, cb_source_format, "format",
+	_("default source reference-format, may be one of: " CB_SF_ALL_NAMES))
```
Comment (GitMensch) [inline cobc/config.def:83]: / source //

File: cobc/cobc.h [hunk 6]
```diff
@@ -92,17 +92,23 @@
...
 	CB_FORMAT_XOPEN_FFF,	/* X/Open Free-form format */
 	CB_FORMAT_ICOBOL_XCARD,	/* ICOBOL xCard */
 	CB_FORMAT_ICOBOL_CRT,	/* ICOBOL Free-form format (CRT) */
 	CB_FORMAT_ACUTERM,	/* ACU Terminal format, named "TERMINAL" */
 	CB_FORMAT_COBOLX,	/* GCOS's COBOLX */
 };
+#define CB_SF_FREE(sf) ((sf) == CB_FORMAT_FREE)
+#define CB_SF_FIXED(sf) ((sf) == CB_FORMAT_FIXED || (sf) == CB_FORMAT_COBOL85)
+
+#define CB_SF_ALL_NAMES							\
```
Comment (GitMensch) [inline cobc/cobc.h:110]: add a comment before that this is used in the help output

File: cobc/cobc.h [hunk 7]
```diff
@@ -92,17 +92,23 @@
...
+	CB_FORMAT_COBOL85,	/* Fixed format with Area A checks (pending) */
+	CB_FORMAT_FREE,		/* COBOL2002+ free-form */
+	CB_FORMAT_VARIABLE,	/* MF's Variable fixed-form format */
 	CB_FORMAT_XOPEN_FFF,	/* X/Open Free-form format */
 	CB_FORMAT_ICOBOL_XCARD,	/* ICOBOL xCard */
 	CB_FORMAT_ICOBOL_CRT,	/* ICOBOL Free-form format (CRT) */
 	CB_FORMAT_ACUTERM,	/* ACU Terminal format, named "TERMINAL" */
 	CB_FORMAT_COBOLX,	/* GCOS's COBOLX */
 };
+#define CB_SF_FREE(sf) ((sf) == CB_FORMAT_FREE)
```
Comment (GitMensch) [inline cobc/cobc.h:107]: `sf` should not be used as an assignment, should it? if not please drop the additional parenthesis which look like "it should".  [I'm possibly the only person in the world that would be confused by that...]

File: cobc/help.c [hunk 8]
```diff
@@ -81,8 +81,9 @@ cobc_print_usage_common_options (void)
 	        "                        acu-strict, acu, rm-strict, rm, gcos-strict,\n"
 	        "                        gcos;\n"
 	        "                        see configuration files in directory config"));
-	puts (_("  -F, --free            use free source format"));
-	puts (_("  --fixed               use fixed source format (default)"));
+	puts (_("  -F, --free            use free source format (alias of -fformat=free)"));
```
Comment (GitMensch) [inline cobc/help.c:84]: not sure, but wouldn't English-correct be "alias for"?

File: config/mf-strict.conf [hunk 9]
```diff
@@ -26,6 +26,9 @@ name: "Micro Focus COBOL"
 # Value: enum (see default.conf and cobc/cobc.h for details)
 standard-define			1
 
+# Default source reference-format (see default.conf for details)
+format:				fixed # CHECKME
```
Comment (GitMensch) [inline config/mf-strict.conf:30]: checked, that's definitive the case, 01 declarations and section names can start in column 13 and code in column 8.

File: doc/ChangeLog [hunk 10]
```diff
@@ -1,4 +1,8 @@
 
+2022-07-28  Nicolas Berthier <nicolas.berthier@ocamlpro.com>
```
Comment (GitMensch) [inline doc/ChangeLog:2]: I suggest to combine this and the next Changelog entry as `-fformat` is new in any case, maybe go with the current date?

[that's no request, more the way I'd do it]

**Review by GitMensch — APPROVED** [review]
please do the minor adjustments and commit upstream

**Comment by nberth** [comment]
now upstream

**Pull request closed without merging**